tooling/nargo/src/foreign_calls/layers.rs

```diff
@@ -122,6 +122,28 @@ impl<T> Layering for T {
     }
 }
 
+/// A case where we can have either this or that type of handler.
+pub enum Either<L, R> {
```

Comment:
We could then just have an optional layer here rather than needing an `Either` case.

Reply:
Yeah, so the same argument went against the optional layer: because I didn't want to silently let calls to mocks in production programs go unhandled, I wanted the programmer to realise what there was something wrong. 

Instead of adding an `Either` I thought about adding an `enabled` flag to the existing mock handler, but somehow this seemed more attractive from an open-closed principle perspective.